apps/vanilla/src/pages/blackjack/ts/core/game.ts

```diff
@@ -0,0 +1,137 @@
+import { ResultDisplayer } from '../views/resultDisplayer';
+
+import { WinnerDisplayer } from '../views/winnerDisplayer';
+
+import { HistoryDisplayer } from '../views/historyDisplayer';
+
+import { getTotalSum } from '../utils/utils';
+
+import { TurnGenerator } from './turnGenerator';
+import { Player } from './player';
+import { DiceGenerator } from './diceGenerator';
+import { Publisher } from './publisher';
+
+/**
+ * Type for Game class constructor.
+ */
+type GameConstructorData = {
+
+	/** The number of players. */
+	playersCount: number;
+
+	/** The number of sides on the dice. */
+	diceSidesCount: number;
+
+	/** Array of HTML elements to display player results. */
+	playerElements: HTMLElement[];
+
+	/** The HTML element to display history of rolls. */
+	historyElement: HTMLElement;
+
+	/** The button to roll the dice. */
+	diceButtonElement: HTMLElement;
+};
+
+/**
+ * Game class that manages the game logic and UI updates.
+ * @param GameConstructorData - Params needed for the Game class instance.
+ */
+export class Game {
+	private readonly players: readonly Player[];
+
+	private readonly turnGenerator$: TurnGenerator;
+
+	private readonly diceGenerator$: DiceGenerator;
+
+	private readonly historyPublisher$: Publisher<number>;
+
+	private readonly historyDisplayer: HistoryDisplayer;
+
+	private isGameEnded: boolean;
+
+	private readonly rollDiceButton: HTMLElement;
+
+	private bondedPlayTurn: () => void;
+
+	public constructor(constructorData: GameConstructorData) {
+		this.players = new Array(constructorData.playersCount).fill(null)
+			.map(() => new Player());
+		this.turnGenerator$ = new TurnGenerator(constructorData.playersCount);
+		this.diceGenerator$ = new DiceGenerator(constructorData.diceSidesCount);
+		this.historyPublisher$ = new Publisher<number>();
+		this.historyDisplayer = new HistoryDisplayer(constructorData.historyElement);
+		this.isGameEnded = false;
+		this.rollDiceButton = constructorData.diceButtonElement;
+
+		/** Bounded playTurn version for add/remove eventlistener on rollDiceButton. */
+		this.bondedPlayTurn = this.playTurn.bind(this);
```

Comment:
```suggestion
		this.bondedPlayTurn = this.playTurn.bind(this);
```
It's strange to use another method to bind a class method that already exists in this class. Let's use direct binding